lib/method_profiler.rb

```diff
@@ -38,6 +38,56 @@ def #{method_name}(*args, &blk)
     klass.class_eval patches
   end
 
+  def self.patch_with_debug_sql(klass, methods, name, no_recurse: false)
+    patches = methods.map do |method_name|
+
+      recurse_protection = ""
+      if no_recurse
+        recurse_protection = <<~RUBY
+          return #{method_name}__mp_unpatched_debug_sql(*args, &blk) if @mp_recurse_protect_#{method_name}
+          @mp_recurse_protect_#{method_name} = true
+        RUBY
+      end
+
+      <<~RUBY
+      unless defined?(#{method_name}__mp_unpatched_debug_sql)
+        alias_method :#{method_name}__mp_unpatched_debug_sql, :#{method_name}
+        def #{method_name}(*args, &blk)
+          unless prof = Thread.current[:_method_profiler]
```

Comment:
maybe we don't need any of this storage here? the only goal is to output sugg to the console.